Doc 1585/ttp step 3 sandbox and rename (#586)
swan-io/docs: docs/preview/in-person/cards/sandbox.mdx

---
title: Sandbox
---

# Sandbox: In-person card payments

When testing your integration, use the Partner Testing API to simulate in-person card payment events.

## Prerequisites {#prerequisites}

- Merchant profile status: **Enabled**.
- In-person card payment method status: **Enabled**.
- [Simulate updating the statuses](../../../topics/merchants/sandbox.mdx) for your merchant profile and payment method if needed.

## Simulate enabling the payment method {#enable-method}

Use the existing `simulateMerchantPaymentMethodRequestOutcome` mutation to enable your in-person card payment method.

```graphql
mutation EnableInPersonCardMethod {
  simulateMerchantPaymentMethodRequestOutcome(
    input: {
      merchantPaymentMethodId: "$YOUR_PAYMENT_METHOD_ID"
      merchantPaymentMethodStatus: Enabled
    }
  ) {
    __typename
    ... on Rejection {
      message
    }
  }
}
```

## Simulate a payment authorization {#authorize}

To simulate a card payment, first [create a payment intent](./guide-create-payments.mdx#create-intent) using the main API, then call `simulateIncomingInPersonCardAuthorization` in the Testing API with the `secret` from the payment intent.

> When the simulation runs, Swan creates the merchant payment and underlying transactions as if a real card was tapped.

Reviewer: ```suggestion
During the simulation, Swan generates the merchant payment and related transactions as if a real card were being used.
```
